### PR TITLE
Feature/#557 messaging push notification merge into develop

### PR DESCRIPTION
**Pull request by p-maks**
implemented push notification for conversation messaging

### CODE REVIEW

File: root/api/src/main/java/org/patientview/api/client/FirebaseClient.java [hunk 1]
```diff
@@ -54,33 +56,101 @@ public String push(Long userId) {
...
+
+        try {
+
+            /**
+             * The same firebase app for Test and Live
+             * but different topics per env
+             * LIVE: /topics/{userid}-messages
+             * TEST:  /topics/{userid}-test-messages
+             */
+            String topic = "/topics/" + userId + "-test-messages";
```
Comment (twstuart) [inline root/api/src/main/java/org/patientview/api/client/FirebaseClient.java]: Seems a bit hacky having the same firebase project for Live and Test. 
Is there not a way around this?

File: root/api/src/main/java/org/patientview/api/client/FirebaseClient.java [hunk 2]
```diff
@@ -54,33 +56,101 @@ public String push(Long userId) {
...
+
+        RestTemplate restTemplate = new RestTemplate();
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.set("Authorization", "key=" + SERVER_KEY);
+        httpHeaders.set("Content-Type", "application/json;charset=UTF-8");
+
+        JSONObject body = new JSONObject();
+        body.put("to", topic);
+        body.put("priority", "high");
+        //body.put("dry_run", true); // to test without sending the message
```
Comment (twstuart) [inline root/api/src/main/java/org/patientview/api/client/FirebaseClient.java]: Remove commented out code

File: root/api/src/main/java/org/patientview/api/service/impl/ConversationServiceImpl.java [hunk 3]
```diff
@@ -89,7 +90,7 @@
 
 /**
  * Conversation service, for CRUD operations related to Conversations and Messages.
- *
+ * <p>
```
Comment (twstuart) [inline root/api/src/main/java/org/patientview/api/service/impl/ConversationServiceImpl.java]: Remove or closing </p>

File: root/api/src/main/java/org/patientview/api/client/FirebaseClient.java (hunk 1 above)
Comment (p-maks) [reply root/api/src/main/java/org/patientview/api/client/FirebaseClient.java]: yes, firebase does not support different env or something, this was discussed with kyle and we decided to use this way

**Pull request merged**